xdsl/transforms/jax_use_donated_arguments.py

```diff
@@ -14,60 +14,53 @@
     RewritePattern,
     op_type_rewrite_pattern,
 )
-from xdsl.transforms.mlir_opt import MLIROptPass
+from xdsl.utils.exceptions import VerifyException
 
-"""
-@dataclass
-class SubstituteDonatedTensors(RewritePattern):
-    @op_type_rewrite_pattern
-    def match_and_rewrite(self, op: FuncOp, rewriter: PatternRewriter, /):
-        if op.arg_attrs is None:
-            return
 
-        donated_inputs: list[BlockArgument] = []
-        for inp, attr in zip(op.args, op.arg_attrs):
-            if type(inp.type) is TensorType and "tf.aliasing_output" in attr.data:
-                donated_inputs.append(inp)
-
-        for child_op in op.body.ops:
-            if (
-                hasattr(child_op, "outputs")
-                and type(getattr(child_op, "outputs")) is VarOperand
-            ):
-                value_mapper: dict[SSAValue, SSAValue] = {}
-                for output in getattr(child_op, "outputs"):
-                    for i, arg in enumerate(donated_inputs):
-                        if type(getattr(output, "type")) is TensorType and getattr(
-                            arg, "type"
-                        ).is_same_type_with(output.type):
-                            value_mapper[output] = donated_inputs.pop(i)
-                            break
-                new_op = child_op.clone(value_mapper)
-                rewriter.replace_op(child_op, [new_op])
-"""
+def make_materialize_op(source: SSAValue, dest: SSAValue) -> MaterializeInDestination:
+    return MaterializeInDestination(operands=[source, dest], result_types=[source.type])
 
 
 @dataclass
 class SubstituteDonatedTensors(RewritePattern):
     @op_type_rewrite_pattern
-    def match_and_rewrite(self, op: FuncOp, rewriter: PatternRewriter, /):
-        if op.arg_attrs is None:
+    def match_and_rewrite(self, op: Return, rewriter: PatternRewriter, /):
+        func_op = op.parent_op()
+        if func_op is None:
+            raise VerifyException("Return operation should be tied to a FuncOp")
+
+        arg_attrs = getattr(func_op, "arg_attrs")
+        args = getattr(func_op, "args")
```

Comment:
we try to avoid getattr when possible, could you please check that the parent is a func, and then access properties directly?